pytorch_lightning/trainer/supporters.py

```diff
@@ -170,12 +171,35 @@ def to_disk(self) -> None:
                 torch.save(outputs, fp)
 
 
+@dataclass
+class SharedCycleIteratorState:
+
+    mode: str = "max_size_cycle"
```

Comment:
```suggestion
class SharedCycleIteratorState:
    """A state shared between all CylceIterators in a CombinedLoader. 
    
    With a shared state, the iterators can decide to terminate based on the state of all others.
    If the mode is *max_size_cycle*, all iterators need to have finished before the combined loading is considered
    finished, and otherwise any iterator finishing early will lead to all iterators ending early.
    """
    mode: str = "max_size_cycle"
```

Reply:
What do you think of these docs? Did I get it right?